course/content.py

```diff
@@ -1510,25 +1510,44 @@ def instantiate_flow_page(location, repo, page_desc, commit_sha):
 # }}}
 
 
-def get_course_commit_sha(course, participation):
-    # type: (Course, Optional[Participation]) -> bytes
+class CourseCommitShaDoesNotExist(Exception):
+    pass
+
 
-    # logic duplicated in course.utils.CoursePageContext
+def get_course_commit_sha(course, participation, repo=None,
+                          raise_on_nonexistent_preview_commit=False):
+    # type: (Course, Optional[Participation], Optional[Repo_ish], Optional[bool]) -> bytes  # noqa
```

Comment:
`Optional` is not the correct thing to use here. `Optional` indicates that a value may be `None`, not that it is an optional argument.

Reply:
I knew that, but `repo` here can be `None`.

Reply:
Er, the actual issue was with the inner function defined.